CMakeLists.txt

```diff
@@ -823,6 +826,7 @@ Halt")
     DEPENDS ${YDB_SOURCE_DIR}/sr_unix/mumps.cmake mumps mupip
             ${YDB_BINARY_DIR}/${help}help.in1
             ${YDB_BINARY_DIR}/${help}help.in2
+	    ${m_routines}
```

Comment:
The comment will be something like:

Because Ninja orders things differently than the Unix Makefile generator did, we got unlucky in that the help files happen to get created before the M files are copied from the source directories. Adding a dependency to the help files to ensure the M files are placed before the help routines are generated fixed this issue.

I was unable to get the error message back because rerunning after partially reverting my changes did not cause the problem to reappear; this is chance.

Reply:
Rerunning again got is:

```
chathaway@bender:~/eclipse-workspace/ydb$ ninja
[0/1] Re-running CMake...
-- Eclipse version is set to 3.6 (Helios). Adjust CMAKE_ECLIPSE_VERSION if this is wrong.
--> OS = Linux / ARCH = x86_64
-- Bootstrapping from pre-generated sources.
-- Setting locale to ar_AE.utf8
-- Configuring done
-- Generating done
-- Build files have been written to: /home/chathaway/eclipse-workspace/ydb
[2114/2348] Generating gtmhelp.dat, gtmhelp.gld
FAILED: gtmhelp.dat gtmhelp.gld 
cd /home/chathaway/eclipse-workspace/ydb && /usr/bin/cmake -E remove gtmhelp.dat gtmhelp.gld && /usr/bin/cmake -D ydb_dist=/home/chathaway/eclipse-workspace/ydb -D ydb_routines=. -D ydb_gbldir=/home/chathaway/eclipse-workspace/ydb/gtmhelp -D mumps=/home/chathaway/eclipse-workspace/ydb/mumps -D "args=-run;GDE" -D input_file=/home/chathaway/eclipse-workspace/ydb/gtmhelp.in1 -P /home/chathaway/p/YottaDB/sr_unix/mumps.cmake && /usr/bin/cmake -D ydb_dist=/home/chathaway/eclipse-workspace/ydb -D ydb_routines=. -D ydb_gbldir=/home/chathaway/eclipse-workspace/ydb/gtmhelp -D mumps=/home/chathaway/eclipse-workspace/ydb/mupip -D args=create -P /home/chathaway/p/YottaDB/sr_unix/mumps.cmake && /usr/bin/cmake -D ydb_dist=/home/chathaway/eclipse-workspace/ydb -D ydb_routines=. -D ydb_gbldir=/home/chathaway/eclipse-workspace/ydb/gtmhelp -D mumps=/home/chathaway/eclipse-workspace/ydb/mumps -D args=-direct -D input_file=/home/chathaway/eclipse-workspace/ydb/gtmhelp.in2 -P /home/chathaway/p/YottaDB/sr_unix/mumps.cmake
%YDB-E-ZLINKFILE, Error while zlinking "GDE"
%YDB-E-FILENOTFND, File GDE not found
CMake Error at /home/chathaway/p/YottaDB/sr_unix/mumps.cmake:43 (message):
  Command </home/chathaway/eclipse-workspace/ydb/mumps -run;GDE
  INPUT_FILE;/home/chathaway/eclipse-workspace/ydb/gtmhelp.in1 > failed with
  result ='253'


[2115/2348] Generating gdehelp.dat, gdehelp.gld
FAILED: gdehelp.dat gdehelp.gld 
cd /home/chathaway/eclipse-workspace/ydb && /usr/bin/cmake -E remove gdehelp.dat gdehelp.gld && /usr/bin/cmake -D ydb_dist=/home/chathaway/eclipse-workspace/ydb -D ydb_routines=. -D ydb_gbldir=/home/chathaway/eclipse-workspace/ydb/gdehelp -D mumps=/home/chathaway/eclipse-workspace/ydb/mumps -D "args=-run;GDE" -D input_file=/home/chathaway/eclipse-workspace/ydb/gdehelp.in1 -P /home/chathaway/p/YottaDB/sr_unix/mumps.cmake && /usr/bin/cmake -D ydb_dist=/home/chathaway/eclipse-workspace/ydb -D ydb_routines=. -D ydb_gbldir=/home/chathaway/eclipse-workspace/ydb/gdehelp -D mumps=/home/chathaway/eclipse-workspace/ydb/mupip -D args=create -P /home/chathaway/p/YottaDB/sr_unix/mumps.cmake && /usr/bin/cmake -D ydb_dist=/home/chathaway/eclipse-workspace/ydb -D ydb_routines=. -D ydb_gbldir=/home/chathaway/eclipse-workspace/ydb/gdehelp -D mumps=/home/chathaway/eclipse-workspace/ydb/mumps -D args=-direct -D input_file=/home/chathaway/eclipse-workspace/ydb/gdehelp.in2 -P /home/chathaway/p/YottaDB/sr_unix/mumps.cmake
%YDB-E-ZLINKFILE, Error while zlinking "GDE"
%YDB-E-FILENOTFND, File GDE not found
CMake Error at /home/chathaway/p/YottaDB/sr_unix/mumps.cmake:43 (message):
  Command </home/chathaway/eclipse-workspace/ydb/mumps -run;GDE
  INPUT_FILE;/home/chathaway/eclipse-workspace/ydb/gdehelp.in1 > failed with
  result ='253'


[2116/2348] Generating dsehelp.dat, dsehelp.gld
FAILED: dsehelp.dat dsehelp.gld 
cd /home/chathaway/eclipse-workspace/ydb && /usr/bin/cmake -E remove dsehelp.dat dsehelp.gld && /usr/bin/cmake -D ydb_dist=/home/chathaway/eclipse-workspace/ydb -D ydb_routines=. -D ydb_gbldir=/home/chathaway/eclipse-workspace/ydb/dsehelp -D mumps=/home/chathaway/eclipse-workspace/ydb/mumps -D "args=-run;GDE" -D input_file=/home/chathaway/eclipse-workspace/ydb/dsehelp.in1 -P /home/chathaway/p/YottaDB/sr_unix/mumps.cmake && /usr/bin/cmake -D ydb_dist=/home/chathaway/eclipse-workspace/ydb -D ydb_routines=. -D ydb_gbldir=/home/chathaway/eclipse-workspace/ydb/dsehelp -D mumps=/home/chathaway/eclipse-workspace/ydb/mupip -D args=create -P /home/chathaway/p/YottaDB/sr_unix/mumps.cmake && /usr/bin/cmake -D ydb_dist=/home/chathaway/eclipse-workspace/ydb -D ydb_routines=. -D ydb_gbldir=/home/chathaway/eclipse-workspace/ydb/dsehelp -D mumps=/home/chathaway/eclipse-workspace/ydb/mumps -D args=-direct -D input_file=/home/chathaway/eclipse-workspace/ydb/dsehelp.in2 -P /home/chathaway/p/YottaDB/sr_unix/mumps.cmake
%YDB-E-ZLINKFILE, Error while zlinking "GDE"
%YDB-E-FILENOTFND, File GDE not found
CMake Error at /home/chathaway/p/YottaDB/sr_unix/mumps.cmake:43 (message):
  Command </home/chathaway/eclipse-workspace/ydb/mumps -run;GDE
  INPUT_FILE;/home/chathaway/eclipse-workspace/ydb/dsehelp.in1 > failed with
  result ='253'


[2117/2348] Generating mupiphelp.dat, mupiphelp.gld
FAILED: mupiphelp.dat mupiphelp.gld 
cd /home/chathaway/eclipse-workspace/ydb && /usr/bin/cmake -E remove mupiphelp.dat mupiphelp.gld && /usr/bin/cmake -D ydb_dist=/home/chathaway/eclipse-workspace/ydb -D ydb_routines=. -D ydb_gbldir=/home/chathaway/eclipse-workspace/ydb/mupiphelp -D mumps=/home/chathaway/eclipse-workspace/ydb/mumps -D "args=-run;GDE" -D input_file=/home/chathaway/eclipse-workspace/ydb/mupiphelp.in1 -P /home/chathaway/p/YottaDB/sr_unix/mumps.cmake && /usr/bin/cmake -D ydb_dist=/home/chathaway/eclipse-workspace/ydb -D ydb_routines=. -D ydb_gbldir=/home/chathaway/eclipse-workspace/ydb/mupiphelp -D mumps=/home/chathaway/eclipse-workspace/ydb/mupip -D args=create -P /home/chathaway/p/YottaDB/sr_unix/mumps.cmake && /usr/bin/cmake -D ydb_dist=/home/chathaway/eclipse-workspace/ydb -D ydb_routines=. -D ydb_gbldir=/home/chathaway/eclipse-workspace/ydb/mupiphelp -D mumps=/home/chathaway/eclipse-workspace/ydb/mumps -D args=-direct -D input_file=/home/chathaway/eclipse-workspace/ydb/mupiphelp.in2 -P /home/chathaway/p/YottaDB/sr_unix/mumps.cmake
%YDB-E-ZLINKFILE, Error while zlinking "GDE"
%YDB-E-FILENOTFND, File GDE not found
CMake Error at /home/chathaway/p/YottaDB/sr_unix/mumps.cmake:43 (message):
  Command </home/chathaway/eclipse-workspace/ydb/mumps -run;GDE
  INPUT_FILE;/home/chathaway/eclipse-workspace/ydb/mupiphelp.in1 > failed
  with result ='253'


[2119/2348] Linking C executable gtmsecshrdir/gtmsecshr
ninja: build stopped: subcommand failed.
```

Reply:
Thanks. Adding some parts of the above comment in the commit would help.